### PR TITLE
test for completion participant

### PR DESCRIPTION
**Pull request by octref**

### CODE REVIEW

File: src/test/completionParticipant.test.ts [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+    assert.equal(attributeValueParticipationCount, 1);
+    assert.equal(tag, 'div');
+    assert.equal(attribute, 'class');
+    assert.equal(value, '');
+    assert.equal(range.start.line, 0);
+    assert.equal(range.start.character, 12);
+    assert.equal(range.end.line, 0);
+    assert.equal(range.end.character, 12);
+
+    testCompletionWithParticipantsFor('<div class="foo|"></div>', [participant]);
```
Comment (aeschli) [inline src/test/completionParticipant.test.ts]: What if the cursor is before or after the `"`? What range is returned then?

It seems to me the range should always include the quotes.

Comment (octref) [reply src/test/completionParticipant.test.ts]: I think this is in-line with the `onHtmlAttributeValue` where they only receive the unquoted value, as I feel the participant shouldn't need to worry about the quotes outside.

If you think it's better to have the quotes, I'll change both the test and the `value` in `onHtmlAttributeValue`, and do the extraction of the surrounding quotes in HTML server's `pathCompletion`.

Comment (aeschli) [reply src/test/completionParticipant.test.ts]: I think it's better to always get the full range, including the quotes. That way the completion particpant can also decide to remove the quotes. The 'value' should be without quotes

Comment (octref) [reply src/test/completionParticipant.test.ts]: I did it this way. However the previous behavior is that for `<div class='foo'>`, with `scanner.getTokenText()` you seem to get normalized value with double quotes (`"foo"`). But that's for another issue I guess.

**Comment by aeschli** [comment]
Thanks @octref !

**Pull request merged**